src/main/java/com/ibm/crypto/plus/provider/ock/NativeInterface.java

```diff
@@ -131,140 +131,50 @@ static void preloadJGskit(String libraryToLoad) {
+        if (osName.startsWith("Windows") && osArch.equals("amd64")) {
+            loadFile = new File(jgskitPath, "libjgskit_64.dll");
         } else if (osName.equals("Mac OS X")) {
             loadFile = new File(jgskitPath, "libjgskit.dylib");
         } else {
             loadFile = new File(jgskitPath, "libjgskit.so");
         }
 
-
         boolean jgskitLibraryPreloaded = loadIfExists(loadFile);
         if (jgskitLibraryPreloaded == false) {
-            String exceptionMessage = "Could not load dependent jgskit library";
-
-            if (debug != null) {
-                // Do not use loadFile or libraryName in message in an effort to hide OCK usage
-                // from users
-                //
-                exceptionMessage = "Could not load dependent jgskit library for os.name=" + osName
-                        + ", os.arch=" + osArch;
-            }
-
-            throw new ProviderException(exceptionMessage);
+            throw new ProviderException("Could not load dependent jgskit library for os.name=" + osName
+                        + ", os.arch=" + osArch);
         }
     }
 
-    static void preloadOCK(String libraryToLoad, boolean add64) {
+    static void preloadOCK(String libraryToLoad) {
         osName = System.getProperty("os.name");
         osArch = System.getProperty("os.arch");
         String ockPath = getOCKLoadPath();
         File loadFile = null;
 
         // --------------------------------------------------------------
-        // Determine the OCK library to load
+        // Determine the OCK library to load for a given OS and architecture.
         //
-        // aix32_ppc: lib<ockCoreLibraryName>.so
-        // aix64_ppc: lib<ockCoreLibraryName>_64.so
-        // hpux32_ia64: lib<ockCoreLibraryName>_32.so
-        // hpux64_ia64: lib<ockCoreLibraryName>_64.so
-        // linux-aarch64: lib<ockCoreLibraryName>_64.so
-        // linux31_s390: lib<ockCoreLibraryName>.so
-        // linux32_ppc: lib<ockCoreLibraryName>.so
-        // linux32_x86: lib<ockCoreLibraryName>.so
-        // linux64_ppc: lib<ockCoreLibraryName>_64.so
-        // linux64_ppcle: lib<ockCoreLibraryName>_64.so
-        // linux64_s390: lib<ockCoreLibraryName>_64.so
-        // linux64_x86: lib<ockCoreLibraryName>_64.so
-        // osx_fat: lib<ockCoreLibraryName>.dylib
-        // sun32_sparc: lib<ockCoreLibraryName>.so
-        // sun32_x86: lib<ockCoreLibraryName>.so
-        // sun64_sparc: lib<ockCoreLibraryName>_64.so
-        // sun64_x86: lib<ockCoreLibraryName>_64.so
-        // win32_x86: <ockCoreLibraryName>.dll
-        // win64_x86: <ockCoreLibraryName>_64.dll
+        // AIX: lib<libraryToLoad>_64.so
+        // Linux aarch64: lib<libraryToLoad>.dylib
+        // Linux ppc64le: lib<libraryToLoad>_64.so
+        // Linux s390x: lib<libraryToLoad>_64.so
+        // Linux x86_64: lib<libraryToLoad>_64.so
+        // Mac OS X: lib<libraryToLoad>_64.so
+        // Windows* amd64: <libraryToLoad>_64.dll
```

Comment:
Is there a reason for the `*` after `Windows`?

Reply:
The only reason I put `*` in this spot is that the code currently checks for the os name to start with Windows, so this is already represented by the comment where we use `Windows*`

Reply:
Oh, so with the `*` you mean that it begins with `Windows`, but there might be other stuff after. Makes sense.